Flaky test `TestMergeRequiredContextsCommitStatus`

https://github.com/go-gitea/gitea/actions/runs/8199270074/job/22424432087

```
--- FAIL: TestMergeRequiredContextsCommitStatus (0.00s)
    commit_status_test.go:62: 
        	Error Trace:	/home/runner/work/gitea/gitea/services/pull/commit_status_test.go:62
        	Error:      	Test case failed
        	Test:       	TestMergeRequiredContextsCommitStatus
        	Messages:   	Test case 4 failed
FAIL
```

--- a/services/pull/commit_status.go
+++ b/services/pull/commit_status.go
@@ -59,7 +59,7 @@ func MergeRequiredContextsCommitStatus(commitStatuses []*git_model.CommitStatus,
 		}
 	}
 
-	if matchedCount == 0 {
+	if matchedCount == 0 && returnedStatus == structs.CommitStatusSuccess {
 		status := git_model.CalcCommitStatus(commitStatuses)
 		if status != nil {
 			return status.State

